lib/containers/utils.pm

```diff
@@ -0,0 +1,163 @@
+    #   - all local images
+    my $local_images_list = script_output("$runtime image ls");
+    die("$runtime image opensuse/tumbleweed not found") unless ($local_images_list =~ /opensuse\/tumbleweed\s*latest/);
+    die("$runtime image opensuse/leap not found")       unless ($local_images_list =~ /opensuse\/leap\s*latest/);
+
+    # Containers can be spawned
+    #   - using 'run'
+    assert_script_run("$runtime container run --name test_1 hello-world | grep 'Hello from Docker\!'");
+    #   - using 'create', 'start' and 'logs' (background container)
+    assert_script_run("$runtime container create --name test_2 alpine:$alpine_image_version /bin/echo Hello world");
+    assert_script_run("$runtime container start test_2 | grep test_2");
+    assert_script_run("$runtime container logs test_2 | grep 'Hello world'");
+    #   - using 'run --rm'
+    assert_script_run(qq{$runtime container run --name test_ephemeral --rm alpine:$alpine_image_version /bin/echo Hello world | grep "Hello world"});
+    #   - using 'run -d' and 'inspect' (background container)
+    my $container_name = 'tw';
+    assert_script_run("$runtime container run -d --name $container_name opensuse/tumbleweed tail -f /dev/null");
+    assert_script_run("$runtime container inspect --format='{{.State.Running}}' $container_name | grep true");
+    my $output_containers = script_output("$runtime container ls -a");
+    die('error: missing container test_1') unless ($output_containers =~ m/test_1/);
+    die('error: missing container test_2') unless ($output_containers =~ m/test_2/);
+    die('error: ephemeral container was not removed') if ($output_containers =~ m/test_ephemeral/);
+    die("error: missing container $container_name") unless ($output_containers =~ m/$container_name/);
+
+    # Containers' state can be saved to a docker image
+    my $exit_code = script_run("$runtime container exec $container_name zypper -n in curl", 300);
+    if ($exit_code) {
+        record_info('poo#40958 - curl install failure, try with force-resolution.');
+        my $output = script_output("$runtime container exec $container_name zypper in --force-resolution -y -n curl", 600);
+        die('error: curl not installed in the container') unless ($output =~ m/Installing: curl.*done/);
+    }
+    assert_script_run("$runtime container commit $container_name tw:saved");
+
+    # Network is working inside of the containers
+    my $output = script_output("$runtime container run tw:saved curl -I google.de");
+    die("network is not working inside of the container tw:saved") unless ($output =~ m{Location: http://www\.google\.de/});
+
+    # Using an init process as PID 1
+    assert_script_run "$runtime run --rm --init opensuse/tumbleweed ps --no-headers -xo 'pid args' | grep '1 .*init'";
+
+    if (script_run('command -v man') == 0) {
+        assert_script_run("man -P cat $runtime build | grep '$runtime-build - Build'");
+    }
```

Comment:
JeOS does not pull docs/man-pages. It has set option `rpm.install.excludedocs=yes` in zypp.conf.

Reply:
I think that the condition here, covers this case. As JeOS does not pull man-pages : `command -v man = 1`, so it will never execute this test.

Reply:
Ah, I have stopped reading after `man`....sry! xD